core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -445,23 +456,45 @@ public PersistenceState getPersistenceState(long fileId) throws FileDoesNotExist
    * {@link FileSystemAction#EXECUTE} permission on the path if it is a directory.
    *
    * @param path the path to get the {@link FileInfo} list for
+   * @param loadDirectChildren whether to load the direct children if path is a directory if its
+   *        direct children have not beed loaded before
    * @return the list of {@link FileInfo}s
    * @throws AccessControlException if permission checking fails
    * @throws FileDoesNotExistException if the file does not exist
    * @throws InvalidPathException if the path is invalid
    */
-  public List<FileInfo> getFileInfoList(AlluxioURI path)
+  public List<FileInfo> getFileInfoList(AlluxioURI path, boolean loadDirectChildren)
```

Comment:
optional: consider creating GetFileInfoListOptions so that in the future adding arguments to this function does not require changing all the call site ... or at least create a TODO for that